docs/cli/ilab-model-management-standardization.md

```diff
@@ -0,0 +1,81 @@
+
+In addition to this, we currently support multiple sources for model downloads. There could be models that span multiple sources and also have different versions and tags that users may want to be able to work with. Currently, there is a lack of uniformity in how we organize models downloaded from Hugging Face vs OCI registries. We do not have a way to differentiate and maintain unique identities for models in cache, based on their sources. We also currently don't have the ability to support version management. Users are only able to work with one version of a given model at a time, as downloading a different version just overwrites their existing model.
+
+## Proposal
+
+### Part 1: Narrowing down to a singular model related flag
+
+#### Problem Statement - Section 1
+
+We want to streamline the number of model related flags available, while establishing consistency and predictability between their uses and handling 3 separate use-cases: absolute paths, relative paths and remote repository names.
+
+#### Suggested approach - Section 1
+
+- Retain `--model` and deprecate all other model related flags
+  - Alternatively, we could retain `--model-path` but passing in a repo name to a flag called "model-path" seems a little more awkward than passing in a path to a field just called "model"
+- `--model` should accept both paths, as well as strings (for repo names)
+- First `--model` should assume whatever is passed into it is an absolute path, and should check against root to determine if that path exists
+  - If it exists, we should run an additional check to determine if it points to a safetensor or gguf models (use existing `is_model_safetensors` and `is_model_gguf` checks for this)
+  - If path does not exist, or exists but is not determined to be a valid model, then move on
+- If path is not an absolute path, assume the path supplied is a relative path. Expand it by appending it to `~/.local/cache/instructlab/models` and check if the path exists. If exists and model is found, use it. If not, move on
+- If supplied content is neither an absolute path, nor a relative path - assume it is the name of a remote repo on HF and download it
+  - alternatively, we can error out here and require that user download the model explicitly via `ilab model download`
+
+This would standardize the behavior of the `--model` flag across all the commands that it appears in. There could be a dedicated model resolver function that implements the above described process.
+
+The only exception may be `ilab model download` which contains a `--model-dir` flag, which acts as a sink rather than a source. This flag could stand to benefit from being renamed to `--destination` instead.
```

Comment:
so, alot of these renames might need to wait for a major version bump to maintain backwards compatibility

Reply:
why? I think they can all follow the same timeline 
introduce a `--destination` right now and add the deprecation notice to `--model-dirs` and get rid of all the deprecated flags in the next 1-2 releases